src/components/FeatureWalkthrough.tsx

```diff
@@ -0,0 +1,200 @@
+import { FunctionComponent, useEffect } from 'react'
+
+import classNames from 'classnames'
+import { startCase } from 'lodash'
+import Link from 'next/link'
+
+import { buttonStyle, buttonLocation } from '@data'
+import { IFeatures } from '@interfaces/featureWalkthrough'
+import { IVideoElement } from '@interfaces/video'
```

Comment:
I don't think we should have global/common interfaces unless we're referencing them in multiple places. Since these are scoped to this component only, they should live within the component (like before) or as collocated files beside the component if you'd rather a more file based approach.

Reply:
Is that to just have less files and a smaller bundle size @bretthayes ? I agree, if the interface is used in multiple places it should maybe live in it's own file. That's what my past teams did. What do you think about changing naming convention to `I{Interface-Name}`? I see we're doing that in a few places already, we're just inconsistent with our interface naming.

Reply:
@st0nebraker It's not necessarily tied to bundle size, but more of a standard practice with React. It reduces context switching and scopes related files together in the file tree for a more organized structure. This is compared to having files all over the tree.

I find the naming convention really weird since it's not PascalCase and I would probably prefer it to be a simplified name to what it references. I've seen folks use the `Props` suffix rather than the `I` prefix (ie: `MyComponentProps` vs. `IMyComponent`) and it's a bit cleaner to read.

For reference, AIrbnb has a really nice [JSX style guide](https://airbnb.io/javascript/react/#naming) and there's also [this TS style guide](https://ts.dev/style/) which is based on Google's style guide. They mention [naming conventions](https://ts.dev/style/#naming-style) and advise against the `I{InterfaceName}` naming convention.

Reply:
Also, if you take a look at the UseCases section component for the homepage, I actually just name [the interface](https://github.com/sourcegraph/about/blob/main/src/pages/home/_UseCases.tsx#L15) with the same name as [the component](https://github.com/sourcegraph/about/blob/main/src/pages/home/_UseCases.tsx#L73) for simplicity. Take a look at [this thread](https://stackoverflow.com/questions/49798156/typescript-variable-name-as-interface-type) and the two answers.

Reply:
@bretthayes The `I` prefix is pascal case. Interesting, I didn't know TypeScript recommended against the `I` prefix, I guess my last 3 teams used the wrong naming convention 😆 - We're using this convention in [this file](https://github.com/sourcegraph/about/blob/main/src/hooks/chiliPiper.ts#L15) which is what made me think we didn't have an agreed upon convention. Happy to defer to what the team wants, change the trailing `I`-prefixed interfaces, and reinforce this method going forward, as long as the whole team is clear. Interesting to note the SG mono-repo has inconsistencies with their interface naming as well: [example](https://github.com/sourcegraph/sourcegraph/blob/main/client/web/src/org/members/InviteMemberModal.tsx#L22)

Reply:
I agree! Dually noted 😄 - Keep interface/type names consistent with their variable/component unless exporting to share 👍🏻  Thanks for the sources